x/auth/client/cli/tx_multisign.go

```diff
@@ -1,163 +1,127 @@
-		case sigOnly && !cliCtx.Indent:
-			json, err = cdc.MarshalJSON(newTx.Signatures[0])
-		case !sigOnly && cliCtx.Indent:
-			json, err = cdc.MarshalJSONIndent(newTx, "", "  ")
-		default:
-			json, err = cdc.MarshalJSON(newTx)
-		}
-		if err != nil {
-			return err
-		}
-
-		if viper.GetString(flagOutfile) == "" {
-			fmt.Printf("%s\n", json)
-			return
+	return func(cmd *cobra.Command, args []string) error {
+
+		var fromName = viper.GetString(flags.FlagFrom)
+		var inFileName = args[0]
+		var outFileName = args[1]
+
+		// Get TxBuilder from cli context
+                var keybase, err = keyring.New(sdk.KeyringServiceName(), viper.GetString(flags.FlagKeyringBackend), viper.GetString(flags.FlagHome), os.Stdin)
+                if err != nil {
+                        panic(err)
+                } 
+
+		// Parse StdSignDocs from file.
+		var docs = parseStdSignDocsFromFile(inFileName)
+
+		// Collect signatures.
+		var sigs []types.StdSignature = nil
+		for _, doc := range docs {
+			sig, pubkey, err := keybase.Sign(fromName, doc.Bytes())                                                  
+			if err != nil { panic(err) }
+			sigs = append(sigs, types.StdSignature{
+				PubKey:pubkey.Bytes(),
+				Signature: sig,
+			})
+
+			// TODO: at this point, it is possible to derive the address from the pubkey,
+			// and so we could auto-inject the signature in the appropriate place,
+			// *if* we could also deal with multisig.  But I'm not sure how
+			// best to solve this (ideally the cli flags + the keybase includes enough information
+			// to figure it out based on the GetSigners(), but that isn't available
+			// unless the sdk.Msg is parsed out.
+			// Given that this function is intended to work only on StdSignDoc's w/
+			// json.RawMessage for msgs, there just isn't enough info to do this automatically,
+			// but perhaps there can be a "sig-position" flag.
+			// Until then, or something else, --signature-only is not supported.
 		}
 
+		// Write StdSignatures to file.
 		fp, err := os.OpenFile(
-			viper.GetString(flagOutfile), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644,
+			outFileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644,
 		)
-		if err != nil {
-			return err
-		}
+		if err != nil { panic(err) }
 		defer fp.Close()
 
-		fmt.Fprintf(fp, "%s\n", json)
-
-		return
+		for _, sig := range sigs {
+			sigJSON, err := json.Marshal(sig)
+			if err != nil { panic(err) }
+			fmt.Fprintf(fp, "%X\n", sigJSON)
+		}
+		//fp.Flush()
+		return nil
 	}
 }
 
-func readAndUnmarshalStdSignature(cdc *codec.Codec, filename string) (stdSig types.StdSignature, err error) {
-	var bytes []byte
-	if bytes, err = ioutil.ReadFile(filename); err != nil {
-		return
+//----------------------------------------
+
+func parseStdSignDocsFromFile(filename string) (docs []types.StdSignDoc) {
+	body, err := ioutil.ReadFile(filename)
+	if err != nil { panic(err) }
+	return parseStdSignDocs(string(body))
+}
+
+func parseStdSignDocs(lines string) (docs []types.StdSignDoc) {
```

Comment:
`--generate-only` generates `StdTx`s. This command seems to be expecting `types.StdSignDoc`s instead.
I'm not sure it would work (unless we amend `--generate-only`'s behaviour)